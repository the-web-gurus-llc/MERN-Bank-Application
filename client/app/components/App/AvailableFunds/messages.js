import { defineMessages } from 'react-intl';

export const scope = 'app.components.AvailableFunds';

export default defineMessages({
  availableFunds: {
    id: `${scope}.available.funds`,
    defaultMessage: 'Available Funds',
  },
});
