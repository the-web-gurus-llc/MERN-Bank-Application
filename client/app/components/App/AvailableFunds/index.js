import React, { useEffect } from 'react';
import Trend from 'react-trend';
import { createStructuredSelector } from 'reselect';
import {
  makeSelectAmountMoney,
  makeSelectAccountBalanceHistory,
  makeSelectCurrencyName,
} from 'containers/DashboardPage/selectors';
import { makeSelectIsLoading } from 'providers/LoadingProvider/selectors';
import { useSelector, useDispatch } from 'react-redux';
import { colors } from 'utils';
import Widget from 'components/App/Widget';
import { FormattedMessage } from 'react-intl';
import { GET_AVAILABLE_FUNDS_REQUEST } from 'containers/DashboardPage/constants';
import { getRequestName } from 'helpers';
import { getAvailableFundsAction } from 'containers/DashboardPage/actions';
import messages from './messages';

const stateSelector = createStructuredSelector({
  amountMoney: makeSelectAmountMoney(),
  accountBalanceHistory: makeSelectAccountBalanceHistory(),
  currencyName: makeSelectCurrencyName(),
  isLoading: makeSelectIsLoading(getRequestName(GET_AVAILABLE_FUNDS_REQUEST)),
});

export default function AvailableFunds() {
  const dispatch = useDispatch();
  const {
    amountMoney,
    accountBalanceHistory,
    currencyName,
    isLoading,
  } = useSelector(stateSelector);

  const getAvailableFunds = () => dispatch(getAvailableFundsAction());

  useEffect(() => {
    getAvailableFunds();
  }, []);

  const trend = (
    <Trend
      smooth
      autoDraw
      autoDrawDuration={1500}
      autoDrawEasing="ease-out"
      data={accountBalanceHistory}
      gradient={[`${colors.primaryBlue}`]}
      radius={0}
      strokeWidth={4.0}
      strokeLinecap="butt"
    />
  );

  return (
    <FormattedMessage {...messages.availableFunds}>
      {(title) => (
        <Widget
          isLoading={isLoading}
          svg={trend}
          title={title}
          content={amountMoney}
          unit={currencyName}
        />
      )}
    </FormattedMessage>
  );
}
