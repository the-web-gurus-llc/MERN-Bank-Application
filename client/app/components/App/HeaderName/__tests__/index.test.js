import React from 'react';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import configureStore from 'utils/configureStore';
import AvailableFunds from '../index';

describe('<AvailableFunds />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a AvailableFunds', () => {
    const { container } = render(
      <Provider store={store}>
        <AvailableFunds />
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
