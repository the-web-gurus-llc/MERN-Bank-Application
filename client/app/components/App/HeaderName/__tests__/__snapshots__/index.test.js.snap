// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvailableFunds /> should render a AvailableFunds 1`] = `
<span
  class="HeaderNamestyle__StyledHeaderName-sc-106aplm-0 dgwlDT"
>
  <div>
    Home
  </div>
</span>
`;
